### PR TITLE
LF-42179 Port patch file changes to repo

### PR DESCRIPTION
**Pull request by grantroch**
Moves changes from build scripts pandas-1.0.5.post1.patch file to repo.

### CODE REVIEW

File: pandas/_libs/src/parser/tokenizer.c [hunk 1]
```diff
@@ -1148,13 +1160,17 @@ static int parser_handle_eof(parser_t *self) {
 
         case ESCAPE_IN_QUOTED_FIELD:
         case IN_QUOTED_FIELD:
+            if (!self->error_bad_lines)
+                return 0;
```
Comment (Tim-Evans-Seequent) [inline pandas/_libs/src/parser/tokenizer.c:1164]: If statements without braces are a common source of bugs, especially related to version control merging. I would prefer to see braces on all if statements.

**Review by Tim-Evans-Seequent — APPROVED** [review]

**Comment by louisbortolozzo** [comment]
It would be good to have a brief explanation of why these changes are needed, either here or in LF-42179.

**Review by louisbortolozzo — APPROVED** [review]

**Comment by grantroch** [comment]
I don't know why these changes were made but all this PR is is taking taking the old patch file and applying it here instead.

**Comment by BenLewis-Seequent** [comment]
That is another reason why patch files aren't great, it's hard to follow the history of the changes. These changes look like they come from https://jira.seequent.com/browse/LF-23200 and https://jira.seequent.com/browse/LF-33643.

**Review by BenLewis-Seequent — APPROVED** [review]
